sql/moz-fx-data-shared-prod/firefox_desktop/newtab_component_content/view.sql

```diff
@@ -0,0 +1,44 @@
+CREATE OR REPLACE VIEW
+  `moz-fx-data-shared-prod.firefox_desktop.newtab_component_content`
+AS
+WITH raw_content_info AS (
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_derived.newtab_component_content_v1`
+),
+-- the purpose of the final CTEs is to apply the necessary UDFs to get rownumber
+content_and_visit_info AS (
+  SELECT
+    raw_content_info.*,
+    mozfun.newtab.determine_grid_layout_v1(
+      section_position IS NOT NULL,
+      app_version,
+      experiments
+    ) AS layout_type,
+  FROM
+    raw_content_info
+),
+add_tiles_per_row AS (
+  SELECT
+    content_and_visit_info.*,
+    mozfun.newtab.determine_tiles_per_row_v1(
+      layout_type,
+      newtab_window_inner_width
+    ) AS num_tiles_per_row
```

Comment:
If you plan to use the visits view `firefox_desktop.newtab_visits_daily`, which in my opinion is a better approach, the udfs used here would be available and recomputing these are not necessary.

Reply:
See: https://github.com/mozilla/bigquery-etl/pull/7988#discussion_r2302122238